### PR TITLE
don't fail on unicode key

### PR DESCRIPTION
**Pull request by snopoke**
@NoahCarnahan

### CODE REVIEW

File: corehq/apps/hqcase/management/commands/ptop_fast_reindexer.py [hunk 1]
```diff
@@ -34,7 +34,12 @@ def view_starting(self, db, view_name, kwargs, total_emitted):
             total_emitted,
             total_emitted + kwargs['limit'] - 1)
         )
-        self.log('  startkey={}, startkey_docid={!r}'.format(kwargs.get('startkey'), kwargs.get('startkey_docid')))
+        startkey = kwargs.get('startkey')
+        if isinstance(startkey, basestring):
+            startkey = startkey.encode('utf8')
```
Comment (millerdev) [inline corehq/apps/hqcase/management/commands/ptop_fast_reindexer.py]: Should say `if isinstance(startkey, unicode):`

File: corehq/apps/hqcase/management/commands/ptop_fast_reindexer.py [hunk 2]
```diff
@@ -34,7 +34,12 @@ def view_starting(self, db, view_name, kwargs, total_emitted):
...
+        elif isinstance(startkey, list):
+            startkey = [i.encode('utf8') for i in startkey if isinstance(i, basestring)]
```
Comment (millerdev) [inline corehq/apps/hqcase/management/commands/ptop_fast_reindexer.py]: Is it your intent to drop all items in `startkey` that are not strings? Again, should only call `encode('utf8')` on unicode objects.

File: corehq/apps/hqcase/management/commands/ptop_fast_reindexer.py [hunk 3]
```diff
@@ -34,7 +34,12 @@ def view_starting(self, db, view_name, kwargs, total_emitted):
...
+        self.log('  startkey={}, startkey_docid={!r}'.format(startkey, kwargs.get('startkey_docid')))
```
Comment (millerdev) [inline corehq/apps/hqcase/management/commands/ptop_fast_reindexer.py]: Would `startkey={!r}` work? That would make it unnecessary to do the encoding above.

Another possibility would be to make the template a unicode string: `u'  startkey={}, startkey_docid={!r}'`

Comment (snopoke) [inline corehq/apps/hqcase/management/commands/ptop_fast_reindexer.py]: https://github.com/dimagi/commcare-hq/pull/10011

**Pull request merged**